src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -104,6 +93,34 @@ public static class ExpressionMetadataProvider
                 metadata = metadataProvider.GetMetadataForType(containerType).Properties[propertyName];
             }
 
+            Func<object, object> modelAccessor = (container) =>
+            {
+                try
+                {
+                    object result = CachedExpressionCompiler.Process(expression)((TModel)container);
+                    if (expression.ReturnType.GetTypeInfo().IsEnum)
```

Comment:
No need for this check to be inside the lambda -- `expression` means it's already capturing. So calculate `var isEnum` once and use it here.